controlplane/kubeadm/internal/controllers/controller_test.go

```diff
@@ -165,7 +165,7 @@ func TestReconcileReturnErrorWhenOwnerClusterIsMissing(t *testing.T) {
 	g.Eventually(func() error {
 		_, err := r.Reconcile(ctx, ctrl.Request{NamespacedName: util.ObjectKey(kcp)})
 		return err
-	}, 10*time.Second).Should(HaveOccurred())
+	}, 10*time.Second).Should(BeNil())
```

Comment:
Let's keep using HaveOccurred for errors (we are doing this almost consistently)